Src/Plugins/Security/SEC101_005.SlackTokenValidator.cs

```diff
@@ -66,40 +68,52 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<
                                                                 Dictionary<string, string> options,
                                                                 ref ResultLevelKind resultLevelKind)
         {
-            var client = new WebClient();
-            var data = new NameValueCollection();
+            const string uri = "https://slack.com/api/auth.test";
 
-            data["token"] = fingerprint.Secret;
+            try
+            {
+                HttpClient client = CreateOrUseCachedHttpClient();
+                var dict = new Dictionary<string, string>
+                {
+                    {"token", fingerprint.Secret },
+                };
 
-            byte[] bytes = client.UploadValues("https://slack.com/api/auth.test",
-                                                  "POST",
-                                                  data);
+                using var request = new HttpRequestMessage(HttpMethod.Post, uri);
+                request.Content = new FormUrlEncodedContent(dict);
 
-            string json = Encoding.UTF8.GetString(bytes);
+                using HttpResponseMessage response = client
+                    .SendAsync(request, HttpCompletionOption.ResponseHeadersRead)
+                    .GetAwaiter()
+                    .GetResult();
 
-            if (json.Contains("invalid_auth"))
-            {
-                return ValidationState.Unauthorized;
-            }
+                string content = response.Content
```

Comment:
>response [](http://example.com/codeflow?start=33&length=8)

should we check if statuscode == okk before checking the content?

Reply:
Not necessary. This (and most other) Slack endpoints only return a 200 status code. You may hit a 5XX if Slack itself is having issues, but if the service is up 200 is the only thing you'll get.

Reply:
Added a quick tet for non-200 return codes just in case we get a 500 for something.